File: app/src/main/java/com/mobile/hinde/connection/Image_URL.java
package com.mobile.hinde.connection;

import android.graphics.drawable.Drawable;
import android.os.AsyncTask;
import java.io.InputStream;
import java.net.URL;

public class Image_URL extends AsyncTask<String, Void, Drawable> {

    private AsyncResponse delegate;

    public Image_URL(AsyncResponse delegate){
        this.delegate = delegate;
    }

    @Override
    protected Drawable doInBackground(String... target) {

        try {
            InputStream is = (InputStream)new URL(target[0]).getContent();
            Drawable d = Drawable.createFromStream(is, "src name");
            return d;
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

    @Override
    protected void onPostExecute(Drawable result) {
        delegate.processFinish(result);
    }

}

File: app/src/main/java/com/mobile/hinde/connection/ImageURL.kt
package com.mobile.hinde.connection

import android.graphics.drawable.Drawable
import android.os.AsyncTask
import java.io.InputStream
import java.net.URL

class ImageURL(private val delegate: AsyncResponse): AsyncTask<String, Any, Any>(){

    override fun doInBackground(vararg params: String?): Drawable? {
        val stream = URL(params[0]).content
        if(stream is InputStream){
            return Drawable.createFromStream(stream,"source")
        }
        return null
    }

    override fun onPostExecute(result: Any) {
        delegate.processFinish(result)
    }

}